### ISSUE TITLE
Assessment information bug

### ISSUE THREAD

**Issue by chigi** [dspachos/whaam-mobileapp#4]
The "Gathered observations" field in the Assessment information panel seem to be not updated according to the submitted observations to the server.

**Comment by dspachos** [comment]
The correct way to update information on gathered observation is to get back this number from the server (via a web service, which currently is not exist, but I discussed it with @gianluca78 a while ago, for future implementation). That is because, if two persons on the same time are gathering observations, then the client information would be incorrect. 

In the main page there is the (pull down to update) function, which updates this information. When the web service is ready, the specific assessment will be update with real server data.

**Comment by gianluca78** [comment]
If I have well understood, @chigi discovered that the pull down doesn't work. @chigi can you confirm this?

@dspachos do you prefer 1 method returning an array with both the count of observation sessions and ABCs or two separate methods?

**Comment by dspachos** [comment]
Currently, the pull down is de-activated, but I believe @chigi meant that when the actual observation ends, the next screen does not have updated information about the number of the gathered observations (which is correct).

The solution in my opinion is, when I send data to the server using the POST webservice, the return must be a JSON with the specific assessment object (using phaseID), so I can instantly update the information on the 'Assessments' screen.The same goes for ABC posts. What do you thing?

**Comment by gianluca78** [comment]
@dspachos ok but there is a little decision to make. At this moment you can sent ABCs for multiple users and multiple baselines. If I modify the POST methods, you will be able to send an array of ABCs for just a baseline Id and a specific user (otherwise I should return an array of different baselines/assessments and I think this will be uncomfortable for you). I mean, baselineId and userId (that as a consequence could be removed) will be on an upper level rather than specific ABC data (like sessionId in this moment). Is it ok for you?

**Comment by dspachos** [comment]
Multiple post of ABCs are useful, so instead of changing it why not just send back an array of assessments (just like you do it in the general user-data GET)? Is this make sense to you?

**Comment by gianluca78** [comment]
Yes, no problem at all. I thought it was more uncomfortable for you. So I'll proceed this way! Thanks

**Comment by gianluca78** [comment]
When you post 2 ABCs for the same baseline, the new result of the post method should be something like:

``` json
[
  [
    {
      "phaseId": 1,
      "phase": "Baseline",
      "phaseStartDate": "2015-01-29T00:00:00+0100",
      "phaseEndDate": "2015-02-05T00:00:00+0100",
      "userIsObserver": true,
      "userIsAllowedToTakeABCs": true,
      "observationLengthInMinutes": 45,
      "observationType": "frequency",
      "plannedNumberOfObservations": 8,
      "gatheredNumberOfObservations": 50,
      "gatheredNumberOfABCs": 40
    }
  ],
  [
    {
      "phaseId": 1,
      "phase": "Baseline",
      "phaseStartDate": "2015-01-29T00:00:00+0100",
      "phaseEndDate": "2015-02-05T00:00:00+0100",
      "userIsObserver": true,
      "userIsAllowedToTakeABCs": true,
      "observationLengthInMinutes": 45,
      "observationType": "frequency",
      "plannedNumberOfObservations": 8,
      "gatheredNumberOfObservations": 50,
      "gatheredNumberOfABCs": 41
    }
  ]
]
```

Is it ok for you? Or you prefer to get a file grouped by phaseId with the most updated data?

**Comment by dspachos** [comment]
Wait, I'm a little bit confused: this gatheredNumberOfABCs is 40 in the one and 41 in the other. The truth is that I need only the last one (41 gathered ABCs), to update the ABC screen info. Am I right or is it something I don't get?

**Comment by gianluca78** [comment]
Ok, that's exactly what I'm asking you! :)
I will return just the last most updated count. Thank you!

**Comment by gianluca78** [comment]
Completed. Now the post methods return an array of assessments with the updated results. Let me know if everything is ok!
